### ISSUE TITLE
Enable altitude as an input

### ISSUE THREAD

**Issue by patrickwalton** [DavidArmstrong/WMM_Tinier#1]
I'd love to use try out this library for a satellite application, but I can't figure out how I would modify it accept an input altitude. Any advice on how I could do that?

**Comment by DavidArmstrong** [comment]
The algorithm is based on work done by NOAA, and is documented on their website at: https://www.ngdc.noaa.gov/geomag/WMM/

The stated purpose of the algorithms developed by NOAA is specifically to provide a predicted value of magnetic variation at the Earth's surface.  I'm sure someone, somewhere has a more comprehensive, and complex model, for the environs in low earth orbit as well.  (JPL? GSFC? NASA? DoD?)  Offhand, I don't know where such a thing lives though.  It's a very special use case.

**Comment by patrickwalton** [comment]
I'm pretty sure the WMM is intended for satellite applications as well. I've read several satellite attitude determination papers that use it. Also see page 6 of the state of the magnetic field report linked on the website you gave that discusses the models inclusion of altitudes up to either 490 or 700 km.

Here again altitude is listed as an input, up to 800 km: https://www.ngdc.noaa.gov/geomag/WMM/soft.shtml

But looking at wmm_tiny repo, I see that it was stripped down to be for Earth's surface only.

**Comment by DavidArmstrong** [comment]
Yep. I chose to use that WMM Tiny library as the baseline code for this library, simply to keep the code size down as small as possible. Most Arduino platforms have a memory constraint that one has to work in when designing a sketch. Even this library won't go into a standard Uno board. A Teensy 4.0 board probably has enough space for a more complete implementation, but that leaves out a lot of other platforms.

**Comment by DavidArmstrong** [comment]
Closing since no further comments after more than six months.

**Comment by nhz2** [comment]
I have a more complete implementation that can run on the Uno board here: https://github.com/nhz2/XYZgeomag

Unlike WMM Tiny, where: 

> The intention of this project is to reduce the code space as much as possible at the expense of RAM. This makes it suitable for small embedded processors short on code memory but with 4.2 kBytes of RAM available. The coefficients are compressed in code but are expanded back into their normal format in RAM when the code runs.

I store the coefficients uncompressed in flash, so more flash is used, but less RAM is used. The Uno has 32 KB of Flash and 2 KB of RAM, so this works out better on that board.

**Comment by patrickwalton** [comment]
Very cool! I've starred it and taken note for when we revisit this!
